### PR TITLE
[API] Show computation costs to StructLog transaction trace

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

Added two fields, `computationCostTotal` and `computationCostUsed` in `debug.trace*` APIs.

```
> debug.traceTransaction("0x58c1096e91560a0aae81332241168d1515729d29098e6899dcb22804ef1635a9")
{
  failed: false,
  gas: 21546,
  returnValue: "000000000000000000000000000000000000000000000000000000000000007b",
  structLogs: [{
      computationCostTotal: 150000000,
      computationCostUsed: 120,
      gas: 146,
      gasCost: 3,
      ...
  }, {
      computationCostTotal: 150000000,
      computationCostUsed: 310,
      gas: 143,
      gasCost: 3,
      memory: [],
      op: "DUP1",
      ...
  }, 
  ...
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hyunsooda** [comment]
@blukat29,

Consider removing the `Computation` field from the `StructLog` struct to reduce its size. This value can be derived within the loop as `ComputationCostLimit - OpcodeCC`. Let me know if this approach is acceptable.

**Comment by blukat29** [comment]
@hyunsooda But the `computation` field is not `Limit - opcodeCC`, it's `ComputationCostLimit - ComputationCostSum`. It's inconvenient to sum up all opcodeCCs from the trace.

**Comment by hyunsooda** [comment]
The calculation is equivalent to accumulating from the limit to each step's opcode CC, and this can be accomplished simply by
```diff
+ sum := cc
   for index, trace := range logs {
      if logTimeout {
          return nil, fmt.Errorf("trace logger timeout")
      }
+     sum -= trace.ComputationCost
      formatted[index] = StructLogRes{
```

Initially, I hesitated to implement this change, fearing it might compromise readability. However, I highlighted its value considering the memory-saving benefits, contingent on the number of `StructLog` instances instantiated.

File: blockchain/vm/logger.go [hunk 1]
```diff
@@ -191,7 +193,11 @@ func (l *StructLogger) CaptureState(env *EVM, pc uint64, op OpCode, gas, cost ui
 		storage = l.changedValues[contract.Address()].Copy()
 	}
 	// create a new snapshot of the EVM.
-	log := StructLog{pc, op, gas, cost, mem, memory.Len(), stck, storage, depth, env.StateDB.GetRefund(), err}
+	var (
+		availableCC = env.Config.ComputationCostLimit - env.opcodeComputationCostSum
+		opcodeCC    = env.interpreter.cfg.JumpTable[op].computationCost
```
Comment (blukat29) [inline blockchain/vm/logger.go:198]:
```suggestion
		remainingComputationCost = env.Config.ComputationCostLimit - env.opcodeComputationCostSum
		opcodeComputationCost    = env.interpreter.cfg.JumpTable[op].computationCost
```

File: api/api_public_blockchain.go [hunk 2]
```diff
@@ -473,19 +473,21 @@ func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Addre
...
+	Stack           *[]string          `json:"stack,omitempty"`
+	Memory          *[]string          `json:"memory,omitempty"`
+	Storage         *map[string]string `json:"storage,omitempty"`
+	Computation     uint64             `json:"computation"`
+	ComputationCost uint64             `json:"computationCost"`
 }
 
 // formatLogs formats EVM returned structured logs for json output
-func FormatLogs(timeout time.Duration, logs []vm.StructLog) ([]StructLogRes, error) {
+func FormatLogs(timeout time.Duration, logs []vm.StructLog, cc uint64) ([]StructLogRes, error) {
```
Comment (blukat29) [inline api/api_public_blockchain.go:490]: Is `cc` unused?

**Review by blukat29 — APPROVED** [review]

**Review by sjnam — APPROVED** [review]

**Pull request merged**